runelite-client/src/main/java/net/runelite/client/plugins/thieving/ChestRespawn.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2019, whs
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.thieving;
+
+import java.time.Instant;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import net.runelite.api.coords.WorldPoint;
+
+@AllArgsConstructor
+@Getter(AccessLevel.PACKAGE)
+class ChestRespawn
+{
+	private final Chest chest;
+	private final WorldPoint worldPoint;
+	private final Instant startTime;
+	private final int respawnTime;
```

Comment:
this shouldn't be needed if there's a reference to the chest type

Reply:
The chest has respawn time in `Duration` object (I'm not sure why Rock needed it that way - maybe because this allow using `Duration.ofMinutes` for readability?). Storing the respawnTime in int avoid calling `.toMillis()` every frames.

Reply:
That's why I think keeping end time is better:

```
Instant now = Instant.now();
if (now.isAfter(endTime) {
   // remove chest
   continue;
}

float ratio = now.until(endTime, ChronoUnit.MILLIS) / respawnDuration;

Reply:
Seems that you still need to call toMillis()?

`float percent = now.until(chestRespawn.getEndTime(), ChronoUnit.MILLIS) / (float) chestRespawn.getChest().getRespawnTime().toMillis();`

Reply:
I mean, earlier we were redesigning to optimize out config getters from render. I'd think that multiplication operation would be much more expensive than getters and should be avoided.